karabo_bridge/simulation.py

```diff
@@ -44,21 +44,37 @@
 # _SHAPE = (_PULSES, _MODULES, _MOD_X, _MOD_Y)
 
 
-def gen_combined_detector_data(source, tid_counter):
+def gen_combined_detector_data(source, tid_counter, corrected=False):
     gen = {source: {}}
 
     # metadata
     ts = time()
     sec, frac = str(ts).split('.')
-    gen[source]['metadata'] = {
-        'source': source,
-        'timestamp': ts,
-        'timestamp.tid': tid_counter,
-        'timestamp.sec': sec,
-        'timestamp.frac': frac.ljust(18, '0')  # attosecond resolution
+    gen[source] = {
+        'metadata.source': source,
+        'metadata.timestamp': ts,
+        'metadata.timestamp.tid': tid_counter,
+        'metadata.timestamp.sec': sec,
+        'metadata.timestamp.frac': frac.ljust(18, '0')  # attosecond resolution
     }
 
     # detector random data
+    if corrected:
+        global _PULSES
+        _PULSES = 31
+        global _SHAPE
+        _SHAPE = (_PULSES, _MODULES, _MOD_X, _MOD_Y)
```

Comment:
I'd like to get rid of the mutable global state at some point, but it doesn't have to be in this PR.